### PR TITLE
Add pizza component

### PR DESCRIPTION
**Pull request by YoniX44**
added orders managment!!!

### CODE REVIEW

File: src/Components/Pizza/Pizza.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+} from "@material-ui/core";
+
+const Pizza = ({ parentUser }) => {
+  const [participants, setParticipants] = useState({});
+
+  useEffect(() => {
+    let newParticipant = {};
+    const newName =
+      Object.keys(participants).length + 1 + ". " + parentUser.displayName;
+    newParticipant[newName] = { slices: 8, toppings: ["tuna"] };
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:20]: Default 4, export to const!

File: src/Components/Pizza/Pizza.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                onChange={(e) => {
+                  let newParticipants = { ...participants };
+                  newParticipants[participant[0]].slices = e.target.value;
+                  setParticipants({ ...newParticipants });
+                }}
+              >
+                <MenuItem value={2}>2</MenuItem>
+                <MenuItem value={4}>4</MenuItem>
+                <MenuItem value={6}>6</MenuItem>
+                <MenuItem value={8}>8</MenuItem>
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:52]: Export to const and map the toppings

File: src/Components/Pizza/Pizza.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        return (
+          <>
+            <Typography>{participant[0]}</Typography>
+            gets
+            <FormControl required>
+              <InputLabel>Slices</InputLabel>
+              <Select
+                value={participants[participant[0]].slices}
+                onChange={(e) => {
+                  let newParticipants = { ...participants };
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:44]: Export to funciton!

File: src/Components/Pizza/Pizza.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                            0,
+                            index
+                          ),
+                          e.target.value,
+                          ...participants[participant[0]].toppings.slice(
+                            index + 1
+                          ),
+                        ];
+                        newParticipants[participant[0]].toppings = newToppings;
+                        setParticipants({ ...newParticipants });
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:76]: Export to function!

File: src/Components/Pizza/Pizza.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            </FormControl>
+            slices and with toppings:
+            {participants[participant[0]].toppings.map((topping, index) => {
+              return (
+                <>
+                  <FormControl required>
+                    <InputLabel>Topping</InputLabel>
+                    <Select
+                      value={topping}
+                      onChange={(e) => {
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:63]: You are not using 'e' it is possible to delete it

File: src/Components/Pizza/Pizza.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            <IconButton
+              onClick={() => {
+                let newParticipants = { ...participants };
+                const newToppings = [
+                  ...participants[participant[0]].toppings,
+                  "tuna",
+                ];
+                newParticipants[participant[0]].toppings = newToppings;
+                setParticipants({ ...newParticipants });
+              }}
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:100]: Export to function

File: src/Components/Pizza/Pizza.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+            >
+              {" "}
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:102]: Delete this line

File: src/Components/Pizza/Pizza.js (hunk 1 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: finished

File: src/Components/Pizza/Pizza.js (hunk 2 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: finished

File: src/Components/Pizza/Pizza.js (hunk 3 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: not finished yet

File: src/Components/Pizza/Pizza.js (hunk 4 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: not finished yet

File: src/Components/Pizza/Pizza.js (hunk 5 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:85]: i am using but will export to function

File: src/Components/Pizza/Pizza.js (hunk 6 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: not finished yet

File: src/Components/Pizza/Pizza.js (hunk 7 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: This line is made automaticly with Prettier

File: src/Components/Pizza/Pizza.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from "react";
+import pizzaToppings from "../../Constants/pizzaTopping";
+import pizzaSlices from "../../Constants/pizzaSlices";
+import AddIcon from "@material-ui/icons/Add";
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:4]: Merge icon imports

File: src/Components/Pizza/Pizza.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+              {participantOrder[0] === parentUser.uid ? (
+                <button onClick={addOrder}>add order</button>
+              ) : (
+                <></>
+              )}
+            </Typography>
+            {participantOrder[1].orders.map((order, index) => {
+              return (
+                <>
+                  {index !== 0 ? (
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:66]: You can change to:

{index && ( <> ...... </> )}

File: src/Components/Pizza/Pizza.js (hunk 5 above)
Comment (idobarlev) [reply src/Components/Pizza/Pizza.js:85]: Export to function!

File: src/Components/Pizza/Pizza.js [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+                  {participantsOrders[participantOrder[0]].orders[
+                    index
+                  ].toppings.map((toppingOut, indexOfTopping) => {
+                    return (
+                      <>
+                        <FormControl required>
+                          <Select
+                            value={toppingOut}
+                            onChange={(e) => {
+                              let newParticipants = { ...participantsOrders };
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:108]: Export to function!

File: src/Components/Pizza/Pizza.js [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+                        ].slices = e.target.value;
+                        setParticipantsOrders({ ...newParticipantsOrders });
+                      }}
+                    >
+                      {pizzaSlices.map((slice) => (
+                        <MenuItem value={slice}>{slice}</MenuItem>
+                      ))}
+                    </Select>
+                  </FormControl>
+                  slices with toppings:
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:98]: Wrap with <p> or <Typography> tag?

File: src/Components/Pizza/Pizza.js [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+                            }}
+                          >
+                            {pizzaToppings.map((toppingIn) => (
+                              <MenuItem value={toppingIn.key}>
+                                {toppingIn.value}
+                              </MenuItem>
+                            ))}
+                          </Select>
+                        </FormControl>
+                        and
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:124]: wrap with <p> or <Typography> tag

File: src/Components/Pizza/Pizza.js [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+                      </>
+                    );
+                  })}
+                  <IconButton
+                    onClick={() => {
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:129]: Export to function

File: src/Components/Pizza/Pizza.js [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+                          .toppings,
+                        pizzaToppings[0].key,
+                      ];
+                      newParticipantsOrders[participantOrder[0]].orders[
+                        index
+                      ].toppings = newToppings;
+                      setParticipantsOrders({ ...newParticipantsOrders });
+                    }}
+                  >
+                    {" "}
```
Comment (idobarlev) [inline src/Components/Pizza/Pizza.js:142]: Delete?

**Review by idobarlev — COMMENTED** [review]
Export to functions, keep the function accurate one function - one purpose.

File: src/Components/Pizza/Pizza.js (hunk 8 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:4]: finished

File: src/Components/Pizza/Pizza.js (hunk 9 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:66]: the syntax breaks at line 78 at ":"

File: src/Components/Pizza/Pizza.js (hunk 11 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:98]: will fix after component break (with styles file)

File: src/Components/Pizza/Pizza.js (hunk 12 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:124]: will fix after component break (with styles file)

File: src/Components/Pizza/Pizza.js (hunk 3 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: will fix after component break

File: src/Components/Pizza/Pizza.js (hunk 4 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: will fix after component break

File: src/Components/Pizza/Pizza.js (hunk 5 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:85]: will fix after component break

File: src/Components/Pizza/Pizza.js (hunk 6 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js]: will fix after component break

File: src/Components/Pizza/Pizza.js (hunk 13 above)
Comment (YoniX44) [reply src/Components/Pizza/Pizza.js:129]: will fix after component break

**Pull request merged**